src/data_generator/csv_generator.py

```diff
@@ -64,8 +69,7 @@ def generate(self):
             df.to_csv(out_path_spec, compression=compression)
         MPI.COMM_WORLD.Barrier()
         source_out_path_spec = self.storage.get_uri(self._file_list[0])
-        for i in range(1, int(self.total_files_to_generate)):
-            if i % self.comm_size == self.my_rank:
-                progress(i+1, self.total_files_to_generate, "Generating CSV Data")
-                out_path_spec = self.storage.get_uri(self._file_list[i])
-                copyfile(source_out_path_spec, out_path_spec)
+        for i in dlp.iter(range(self.my_rank, int(self.total_files_to_generate), self.comm_size)):
```

Comment:
The test is failing due to this. It's copying the same file. u have to skip that file for rank 0 as its already created.